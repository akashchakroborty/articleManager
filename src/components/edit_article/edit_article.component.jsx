import React, { useState, useEffect } from "react";
import { connect } from "react-redux";
import getCurrentData from "../../utils/getCurrentData";
import getReadTimeInMin from "../../utils/getReadTimeInMin";
import { withRouter } from "react-router-dom";
import { editArticle } from "../../redux/articles/articles.action";
import FormComponent from "../form/form.component";

export const EditArticle = ({ article, history, editArticle }) => {
  const {
    title: editTitle,
    description: editDescription,
    author: editAuthor,
    content: editContent,
    id: editId,
    img: editImg,
    avatar: editAvatar,
    claps: editClaps,
  } = article;

  const [changedArticle, setarticle] = useState({
    title: editTitle,
    claps: editClaps,
    description: editDescription,
    id: editId,
    readTime: "0 min read",
    date: getCurrentData(),
    author: editAuthor,
    content: editContent,
    img: editImg,
    avatar: editAvatar,
  });

  const { title, description, author, content, id } = changedArticle;

  useEffect(() => {
    const articleReadTime = getReadTimeInMin(content);

    setarticle({ ...changedArticle, readTime: articleReadTime });
  }, [getReadTimeInMin(content)]);

  const handleSubmit = (event) => {
    event.preventDefault();
    console.log("Submitted");
    editArticle(id, changedArticle);
    history.push("/");
  };

  const handleChange = (event) => {
    event.preventDefault();

    const { value, name } = event.target;

    setarticle({ ...changedArticle, [name]: value });
  };

  return (
    <FormComponent
      handleSubmit={handleSubmit}
      title={title}
      description={description}
      author={author}
      edit
      content={content}
      handleChange={handleChange}
    />
  );
};

export default withRouter(connect(null, { editArticle })(EditArticle));
