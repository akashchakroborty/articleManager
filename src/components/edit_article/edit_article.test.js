import React from "react";
import { mount } from "enzyme";
import { EditArticle } from "./edit_article.component";

describe("EditArticle component", () => {
  let wrapper;
  let mockEditArticle;
  let mockHistory;

  beforeEach(() => {
    mockEditArticle = jest.fn();
    mockHistory = {
      push: jest.fn(),
    };
    const mockProps = {
      article: {
        title: "Mock data",
        description: "Mock data",
        author: "Mock data",
        content: "Mock data",
        id: "Mock data",
        img: "Mock data",
        avatar: "Mock data",
        claps: "Mock data",
      },
      editArticle: mockEditArticle,
      history: mockHistory,
    };

    wrapper = mount(<EditArticle {...mockProps} />);
  });

  it("should render EditArticle component", () => {
    expect(wrapper).toMatchSnapshot();
  });

  it("should call handleChange method when FormComponent changes", () => {
    const value = "hey";
    wrapper
      .find("input")
      .at(0)
      .simulate("change", {
        target: { value, name: "title" },
      });
    expect(wrapper.find("input").at(0).instance().value).toEqual(value);
  });

  it("should call handleSubmit method when FormComponent is submitted", () => {
    const logSpy = jest.spyOn(console, "log");
    wrapper.find("form").at(0).simulate("submit");
    expect(logSpy).toHaveBeenCalledWith("Submitted");
  });
});
