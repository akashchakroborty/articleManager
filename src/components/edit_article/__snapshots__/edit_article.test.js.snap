// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`EditArticle component should render EditArticle component 1`] = `ReactWrapper {}`;
